engine/table/src/main/java/io/deephaven/engine/table/impl/util/JobScheduler.java

```diff
@@ -180,13 +180,23 @@ private void onTaskError(@NotNull final Exception e) {
         protected void onReferenceCountAtZero() {
             final Exception localException = exception.get();
             if (localException != null) {
-                onError.accept(localException);
+                try {
+                    onError.accept(localException);
+                } catch (Exception e) {
+                    e.initCause(localException);
```

Comment:
It says this can only be called once, so if we have a throwable with a cause from accept; will this defeat the point?  Perhaps we should log "localException" and report "e".

Reply:
Used addSuppressed at your suggestion. Good catch!